packages/smooth_app/lib/cache/files/files_cache_impl.dart

```diff
@@ -0,0 +1,129 @@
+// ignore_for_file: avoid_slow_async_io
+part of 'files_cache.dart';
+
+/// Implementation based on [getTemporaryDirectory]
+class _ShortLivingFileCacheManagerImpl extends _FileCacheManagerImpl {
+  _ShortLivingFileCacheManagerImpl({
+    required String subFolderName,
+  }) : super._(
+          rootDirectory: getTemporaryDirectory(),
+          subFolderName: subFolderName,
+        );
+}
+
+/// Implementation based on [getApplicationDocumentsDirectory]
+class _LongLivingFilesCacheManagerImpl extends _FileCacheManagerImpl {
+  _LongLivingFilesCacheManagerImpl({
+    required String subFolderName,
+  }) : super._(
+          rootDirectory: getApplicationDocumentsDirectory(),
+          subFolderName: subFolderName,
+        );
+}
```

Comment:
We don't need these guys. We just create two instances of class `FileCacheManagerImpl`, with different root directories.